spec/services/mapbox/get_user_coordinate_request_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Mapbox::GetUserCoordinateRequest do
+  let(:search_param) { 'phuong 3, quan go vap' }
+  let(:overlimited_search_param) do
+    'iewxtjatiewudcebvhrpqrlxjczsjwnfeodbmltolvzjaifoapiotmcdzyiemstyfskexkwadyxbmvvvuzrserjpmbvakbhkhptyxmysgflvywdbm
+bcdbznfuhwjwcxumogknjbrquksgrhzkelgsvatowikujkxebkyafovpklayiyqkmswtryawaokwiilxtyowkengkoialbqpirxpqysnlufhqirwqnnquy
+nrmfzaspvmllgrhyuyiscpfmtkztqctsvxbpyoopubvgprywfjhwzfhhwgbbhhjeelrmejmwgzkencuimqgwdwyjeidqfonjwjlpdkjamtnjclbocdzjeh
+drudumlxhwmcxsfhnytghskqjagymhmazxekpxvdopdzdfkorvcfwbstrybdfsottbczeylcoilgyonpdihfhosimrcqswwsolzkhtdqubzpoksqggsr
+ikqqzprgqsbjsevpcaxcnzu'
+  end
+
+  describe '#call' do
+    context 'when the request is successfully' do
+      subject(:request) { described_class.new(search_param:) }
+
+      it 'returns correct value' do
+        VCR.use_cassette('mapbox/get_user_coordinate_successfully') do
+          request.call
+
+          expect(request.success?).to eq true
+        end
+      end
+    end
+
+    context 'when the request fails' do
+      subject(:request) { described_class.new(search_param: overlimited_search_param) }
+
+      it 'adds an error message' do
+        VCR.use_cassette('mapbox/get_user_coordinate_unsuccessfully') do
```

Comment:
unecessary used of VCR here, this will raise error before it make any request, and this error may not come from faraday but I did not see you try to handle this error in the code

Reply:
At least my solution is to remove this VCR context, for the next idea I think that we have add_error just to return message to handle. Feel free to discuss